### ISSUE TITLE
Ligand-based progress tracking

### ISSUE THREAD

**Issue by jaimergp** [openkinome/kinoml#21]
This issue will track all the progress related to getting the core functionality for ligand-based models merged to `master`.

## Core functionality

- [x] DatasetProvider exports to PyTorch (see #8)
  - [ ] We might need specific exports to [PyTorch Geometric (very similar)](https://pytorch-geometric.readthedocs.io/en/latest/notes/create_dataset.html)
- [x] MeasurementType implements accurate observation models (see #8)
- [x] System is composed of typed MolecularComponents: Protein, Ligand, Complexes (see #8)
- [x] Featurizers can operate in batch mode or upon dataset access via DataLoader (see #8)
- [x] Filter invalid records (see note in #8 description)
- [ ] Cross validation

## Standard practices

- [ ] Ensure documentation is accurate and up-to-date
  - [ ] Drop `mkdocs` for `Sphinx` + material theme?
- [ ] Unit tests have good coverage and are passing

##  Consolidate previous PRs

- [ ] Bring scripts from #6 
- [x] Bring observation model notebook from #14 

## Pending scientific questions

- [ ] Unknowns.  These could be nuisance parameters in the future, but for now:
  - [ ] `[substrate]`: we could estimate by cross-validation, use relative pIC50s, or add that as a nuisance parameter in the future
  - [ ] uncertainties: estimate per dataset (like [Kramer's paper](https://pubmed.ncbi.nlm.nih.gov/22643060/)) or measurement class
- [ ] Verify % displacement equations. See [discussion](https://github.com/openkinome/kinoml/pull/22#discussion_r479985975)

**Comment by t-kimber** [comment]
Cross-validation in _Core functionality_ could be solved using [skorch](https://skorch.readthedocs.io/en/stable/index.html) package. 
Check PR #27 for some toy example using skorch **with** observation model :)

**Comment by t-kimber** [comment]
A few aspects which still need to be addressed:

- [ ] When dealing with the XGBoost package, there should be a distinction between the observation model and the custom loss. See [this notebook](https://github.com/openkinome/kinoml/blob/api/examples/PKIS2_kinase_informed.ipynb) for more details. 

- For the full detailed explanation of the observation models, losses, custom losses, and associated differentiation (gradient and hessian), please refer to the overleaf document (maybe useful @jaimergp ). 

- [ ] If we want to have a ligand-based only model, we can pass `None` as a kinase featurizer. However, this will not reflect a _real_ ligand-based model, since all the systems would be used in this case. We should instead create _one single_ model for each of the kinases, meaning that if there are **N** unique kinases in a data set, **N** models have to trained. -> This will probably be addressed by looping over all kinases. 

- [ ] Create toy example for dealing with multiple measurement types @t-kimber .

**Comment by schallerdavid** [comment]
Outdated, will create new issues when needed.
